app/controllers/incomeSources/add/AddIncomeSourceStartDateCheckController.scala

```diff
@@ -58,106 +58,94 @@ class AddIncomeSourceStartDateCheckController @Inject()(authenticate: Authentica
-    IncomeSourceType.get(incomeSourceKey) match {
-      case Left(ex: Exception) => Logger("application").error(s"[AddIncomeSourceStartDateController][handleShowRequest]: " +
-        s"Failed fulfil submit request: ${ex.getMessage}")
-        Future.successful(getErrorHandler(isAgent).showInternalServerError())
-      case Right(value) =>
-        handleSubmitRequest(
-          incomeSourceType = value,
-          isAgent = isAgent,
-          isUpdate = isChange
-        )
-    }
+
+    handleSubmitRequest(
+      incomeSourceType = incomeSourceType,
+      isAgent = isAgent,
+      isChange = isChange
+    )
   }
 
   private def handleShowRequest(incomeSourceType: IncomeSourceType,
                                 isAgent: Boolean,
-                                isUpdate: Boolean)
+                                isChange: Boolean)
                                (implicit user: MtdItUser[_]): Future[Result] = {
 
     Future.successful(
       if (isEnabled(IncomeSources))
-        (getAndValidateStartDate(incomeSourceType), getCalls(isAgent, isUpdate, incomeSourceType)) match {
-          case (Right(startDate), (backCall, postAction, _)) =>
+        getAndValidateStartDate(incomeSourceType) match {
+          case Right(startDate) =>
             Ok(
               addIncomeSourceStartDateCheckView(
                 isAgent = isAgent,
-                backUrl = backCall.url,
-                postAction = postAction,
+                backUrl = getBackUrl(incomeSourceType, isAgent, isChange),
                 form = form(incomeSourceType.addStartDateCheckMessagesPrefix),
+                postAction = getPostAction(incomeSourceType, isAgent, isChange),
                 incomeSourceStartDate = longDate(startDate.toLocalDate).toLongDate
               )
             )
-          case (Left(ex), _) =>
+          case Left(ex) =>
             Logger("application").error(s"[AddIncomeSourceStartDateCheckController][handleRequest]: " +
               s"Failed to get income source start date from session, reason: ${ex.getMessage}")
-            getErrorHandler(isAgent).showInternalServerError()
+            showInternalServerError(isAgent)
         }
       else Ok(customNotFoundErrorView())
     )
   }
 
   private def handleSubmitRequest(incomeSourceType: IncomeSourceType,
                                   isAgent: Boolean,
-                                  isUpdate: Boolean)
+                                  isChange: Boolean)
                                  (implicit mtdItUser: MtdItUser[_]): Future[Result] = {
 
     val messagesPrefix = incomeSourceType.addStartDateCheckMessagesPrefix
-
     Future.successful(
       if (isEnabled(IncomeSources))
-        (getAndValidateStartDate(incomeSourceType), getCalls(isAgent, isUpdate, incomeSourceType)) match {
-          case (Right(startDate), (backCall, postAction, successCall)) =>
+        getAndValidateStartDate(incomeSourceType) match {
```

Comment:
if we can change signature for this method pls as this seems not logical to return Either[Throwable, String];
Please use => Either[Throwable, LocalDate]